rpc/types_transaction_interfaces.go

```diff
@@ -0,0 +1,70 @@
+package rpc
+
+type AddDeclareTxnInput interface{}
+
+var _ AddDeclareTxnInput = DeclareTxnV1{}
+var _ AddDeclareTxnInput = DeclareTxnV2{}
+
+type EstimateFeeInput interface{}
+
+var _ EstimateFeeInput = InvokeTxnV0{}
+var _ EstimateFeeInput = InvokeTxnV1{}
+var _ EstimateFeeInput = DeployAccountTxn{}
+var _ EstimateFeeInput = DeclareTxnV1{}
+var _ EstimateFeeInput = DeclareTxnV2{}
+
+type Transaction interface {
+	GetType() TransactionType
+}
+
+var _ Transaction = InvokeTxnV0{}
+var _ Transaction = InvokeTxnV1{}
+var _ Transaction = DeclareTxnV1{}
+var _ Transaction = DeclareTxnV2{}
+var _ Transaction = DeployTxn{}
+var _ Transaction = DeployAccountTxn{}
+var _ Transaction = L1HandlerTxn{}
+
+func (tx InvokeTxnV0) GetType() TransactionType {
+	return tx.Type
+}
+
+func (tx InvokeTxnV1) GetType() TransactionType {
+	return tx.Type
+}
+func (tx DeclareTxnV0) GetType() TransactionType {
+	return tx.Type
+}
+func (tx DeclareTxnV1) GetType() TransactionType {
+	return tx.Type
+}
+func (tx DeclareTxnV2) GetType() TransactionType {
+	return tx.Type
+}
+func (tx DeployTxn) GetType() TransactionType {
+	return tx.Type
+}
+func (tx DeployAccountTxn) GetType() TransactionType {
+	return tx.Type
+}
+func (tx L1HandlerTxn) GetType() TransactionType {
+	return tx.Type
+}
+
+// Note: these allow all types to pass, but are to help users of starknet.go
+// understand which types are allowed where.
+
+type InvokeTxnType interface{}
+
+var _ InvokeTxnType = InvokeTxnV0{}
+var _ InvokeTxnType = InvokeTxnV1{}
+
+type DeclareTxnType interface{}
+
+var _ DeclareTxnType = DeclareTxnV0{}
+var _ DeclareTxnType = DeclareTxnV1{}
+var _ DeclareTxnType = DeclareTxnV2{}
```

Comment:
Could you please check is it possible to use nil pointer here:
```
var _ Type = (*SpecificType)(nil)
```

Reply:
Yes, you can use nil pointer here.

In principle, I think you can pass any type in place of DeclareTxnType since it's just an empty interface, so this doesn't constrain what types are allowed. But it's just the best way I can think of to categorise types like DeclareTxnType. Would love to know if there's a better/cleaner way

Reply:
Ah I see, forgot that it's an empty interface. If you want constraint your interfaces with types you can introduce an empty method like:
```
type SomeCommonType interface {
     SomeCommonType()
 }
 
 type Concrete struct{}
 func (c Concrete) SomeCommonType()
 ```
 Not sure that we need it here IMO. One benefit of using pointers instead of structs - no memory taken by these unused structs (but maybe compiler already optimized for that)

Reply:
Unfortunately there's no common field for the transactions.